.github/workflows/CI.yaml

```diff
@@ -0,0 +1,36 @@
+name: CI
+
+on: [push]
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+          cache: gradle
+
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: Check with Gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: check
+
+      - name: Build with Gradle
+        uses: gradle/gradle-build-action@v2
```

Comment:
What's the advantage of using a dedicated action for invoking Gradle commands, compared to just executing `./gradlew build` etc?

Reply:
When there aren't major performance issues it is nicer to have them separated because you can see what failed way faster instead of having to check the logs of the CI.

Reply:
https://github.com/marketplace/actions/gradle-build-action